include/nbl/asset/utils/IShaderCompiler.h

```diff
@@ -397,47 +426,40 @@ class NBL_API2 IShaderCompiler : public core::IReferenceCounted
 				// used for insertions
 				inline bool operator()(const SEntry& lhs, const SEntry& rhs) const
 				{
-					return lhs.compilerData == rhs.compilerData && lhs.mainFileContents == rhs.mainFileContents;
+					return lhs.compilerArgs == rhs.compilerArgs && lhs.mainFileContents == rhs.mainFileContents;
 				}
 				
-			};
-
-			
-
-			// json serialization methods for the unordered multiset go here since I need access to Hash and KeyEquals 
+			};	
 
 			core::unordered_multiset<SEntry, Hash, KeyEqual> m_container;
-			// If provided at creation time, this will hold the json representing entries in m_container + all SPIRV shaders
-			core::smart_refctd_ptr<ICPUBuffer> storageBuffer = nullptr;
-			// Specifies the size in bytes of the container json. This is so we know the offset into the buffer where shader bytecode starts
-			uint64_t containerJsonSize = 0;
 		};
 
-		inline core::smart_refctd_ptr<ICPUShader> compileToSPIRV(const std::string_view code, const SCompilerOptions& options, core::smart_refctd_ptr<CCache> cache = nullptr) const
+		inline core::smart_refctd_ptr<ICPUShader> compileToSPIRV(const std::string_view code, const SCompilerOptions& options, CCache* cache = nullptr) const
 		{
-			/*core::vector<CCache::SEntry::SPreprocessingDependency> dependencies;
-			if (options.cache)
-			{
-				const auto& mainDep = dependencies.emplace_back(code,options.hash());
-				auto found = options.cache->find(mainDep);
+			CCache::SEntry entry;
+			std::vector<CCache::SEntry::SPreprocessingDependency> dependencies;
+			if (cache) {
+				entry = std::move(CCache::SEntry(code, options));
```

Comment:
a bit of C++ knowledge, when you newly create an object its already an `r-value` reference ergo does not need `std::move`